### PR TITLE
feat: allow different prop

### PR DESCRIPTION
**Pull request by kellyjosephprice**
| 🚥 Resolves ISSUE_ID |
| :------------------- |

## 🧰 Changes

Allows the use of the prop: "name".

```
<Variable name="apiKey" />
```

Since this is going to be a "public" component soon, we thought a less repetitive prop would be appropraite?

## 🧬 QA & Testing

- [ ] tests pass?

### CODE REVIEW

File: index.jsx [hunk 1]
```diff
@@ -22,10 +23,15 @@ class Variable extends React.Component {
     this.props.changeSelected(event.target.value);
   }
 
+  getName() {
+    return this.props.name || this.props.variable;
```
Comment (erunion) [inline index.jsx:27]: does this need a third fallback in case `name` and `variable` are both missing?

Comment (erunion) [reply index.jsx:27]: maybe itll be okay because doing `arr[undefined]` just returns `undefined`.

File: index.jsx [hunk 2]
```diff
@@ -22,10 +23,15 @@ class Variable extends React.Component {
...
+  }
+
   getDefault() {
-    const def = this.props.defaults.filter(Boolean).find(d => d.name === this.props.variable) || {};
+    const def = this.props.defaults.filter(Boolean).find(d => d.name === this.getName()) || {};
+
     if (def.default) return def.default;
-    return this.props.variable.toUpperCase();
+    return this.getName().toUpperCase();
```
Comment (erunion) [inline index.jsx:34]: just incase `getName()` returns `undefined`.

```suggestion
    return this.getName()?.toUpperCase() || undefined;
```

**Review by erunion — APPROVED** [review]

File: index.jsx (hunk 1 above)
Comment (kellyjosephprice) [reply index.jsx:27]: It ought to be a runtime or compile time error? Should I throw like a Props missing error?

Comment (erunion) [reply index.jsx:27]: ya that might be a good idea

**Pull request merged**